src/test/java/uk/co/companieshouse/kafka/error/consumer/DltConsumerTest.java

```diff
@@ -1,50 +1,88 @@
 package uk.co.companieshouse.kafka.error.consumer;
 
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.notNullValue;
-import static uk.co.companieshouse.kafka.error.consumer.TestConsumer.TEST_CONSUMER_ID;
-
-import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
-import org.springframework.kafka.core.KafkaTemplate;
-import org.springframework.kafka.test.context.EmbeddedKafka;
-import org.springframework.test.context.ActiveProfiles;
+import org.springframework.kafka.listener.MessageListenerContainer;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.Message;
 import uk.co.companieshouse.kafka.error.config.KafkaErrorProperties;
+import uk.gov.companieshouse.logging.Logger;
 
-@SpringBootTest
-@ActiveProfiles("test")
-@EmbeddedKafka(partitions = 1, brokerProperties = {"listeners=PLAINTEXT://localhost:9092",
-        "port=9092"})
-public class DltConsumerTest {
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-    @Autowired
-    private TestConsumer consumer;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.*;
+import static uk.co.companieshouse.kafka.error.consumer.DltConsumer.CONSUMER_ID;
+import static uk.co.companieshouse.kafka.error.consumer.DltConsumer.REPLAY_HEADER_MILLIS;
+
+@ExtendWith(MockitoExtension.class)
+public class DltConsumerTest {
 
-    @Autowired
-    private KafkaTemplate<String, byte[]> kafkaTemplate;
+    private final static String TOPIC = "topic";
 
-    @Autowired
+    @Mock
     KafkaListenerEndpointRegistry kafkaListenerEndpointRegistry;
 
-    @Autowired
+    @Mock
     KafkaErrorProperties kafkaErrorProperties;
 
+    @Mock
+    Consumer<?, ?> consumer;
+
+    @Mock
+    Logger logger;
+
+    @Mock
+    MessageListenerContainer messageListenerContainer;
+
+    @InjectMocks
+    private DltConsumer dltConsumer;
+
     @Test
     void testConsumer() throws Exception {
+        executeTest(10L, 100L, 1, 0);
+        executeTest(100L, 100L, 0, 1);
```

Comment:
FYI JUnit 5 has a `@ParameterizedTest` annotation which can run the same test with different values and it will show the output (success/fail/assertion errors) per test run.

Reply:
I know - but that creates a new test instance for each test. I want to run these together to test that we don't set the startOffset the second time around.